File: cpp/validPalindrome2.cpp
#include<bits/stdc++.h>
using namespace std;


bool isPalindrome(string s, int i,int j){
    while(i<=j){
        if(s[i]==s[j]){
            i++;
            j--;
        }
        else return false;
    }
    return true;
}

bool validPalindrome(string s){
    int i=0,j=s.size()-1;

    while(i<=j){
        if(s[i]==s[j]){
            i++;
            j--;
        }
        else{
            return isPalindrome(s,i+1,j) || isPalindrome(s,i,j-1);
        }
    }

    return true;
}

int main()
{
    string s; cin >> s;
    cout << validPalindrome(s) << endl;

}

File: js/validPalindrome2.js

let isPalindrome = function(s){
    let str = s;

    s.split("").reverse().join("");
    return s==str;
}

var validPalindrome = function(s) {
    let i=0, j=s.length; 
    
    while(i<=j){
        if(s[i]==s[j]){
            i++;
            j--;
        }
        else{
            return isPalindrome(s.substring(i+1,j)) || isPalindrome(s.substring(i,j-1));
        }
    }
    return true;
};


let s = "deedee"

console.log(validPalindrome(s));